core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -20,10 +20,10 @@ import (
 // Compatible with:
 // - "OffRamp 1.6.0-dev"
 type ExecutePluginCodecV1 struct {
-	extraDataCodec *common.ExtraDataCodec
+	extraDataCodec common.ExtraDataCodecProvider
```

Comment:
This should not be a coded but a provider of a codec for a specific chain, right?

Reply:
no because this this extra data codec is used to decode source chain encoded message, therefore we can't just have a specific chain codec

Reply:
Sure, that's why the suggestion is to host a `provider` here which can provide you this functionality for any chain you request, or errors if doesn't have it.